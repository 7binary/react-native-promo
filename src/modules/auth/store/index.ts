import {
  createAsyncThunk,
  createSelector,
  createSlice,
  PayloadAction,
  Reducer,
} from '@reduxjs/toolkit';
import { ax, Settings } from '@pmk-team/common';
import { AuthState, ApiSettings, LoginParams, LoginResponse } from './types';

const initialState: AuthState = {
  profile_id: null,
  token: null,
  settings: null,
};

// ACTIONS
const getApiSettings = createAsyncThunk('auth/getApiSettings', async () => {
  const res = await ax().post<{settings: ApiSettings}>('api/settings');
  return res.data.settings;
});

const login = createAsyncThunk('auth/login',
  async ({ client, values, onSuccess }: LoginParams) => {
    const res = await client.post<LoginResponse>('api/login', values);
    onSuccess && onSuccess();
    return res.data;
  });

const userLoginState = (state: AuthState, action: PayloadAction<LoginResponse>) => {
  const { token, profile_id } = action.payload;
  Settings.api.jwtToken = token;
  state.token = token;
  state.profile_id = profile_id;
};

// REDUCER
const slice = createSlice({
  name: 'auth',
  initialState,
  extraReducers: {
    [getApiSettings.fulfilled.type]: (state, action: PayloadAction<ApiSettings>) => {
      state.settings = action.payload;
    },
    [login.fulfilled.type]: userLoginState,
  },
  reducers: {
    userLogin: userLoginState,
    userLogout: state => {
      Settings.api.jwtToken = null;
      state.token = null;
      state.profile_id = null;
    },
  },
});
const { userLogin, userLogout } = slice.actions;

// EXPORT
export * from './types';
export * from './useTestFeature';
export const authReducer = slice.reducer as Reducer<typeof initialState>;
export const authActions = {
  userLogin,
  userLogout,
  login,
  getApiSettings,
};

const selectSelf = (rootState: {auth: AuthState}) => rootState.auth;
export const profileIdSelector = createSelector(selectSelf, state => state.profile_id);
export const settingsSelector = createSelector(selectSelf, state => state.settings);
export const tokenSelector = createSelector(selectSelf, state => state.token);
