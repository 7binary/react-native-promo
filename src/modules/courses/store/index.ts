import { createAsyncThunk, createSlice, PayloadAction, Reducer } from '@reduxjs/toolkit';
import FastImage from 'react-native-fast-image';
import { ax } from '@pmk-team/common';
import { Course, CoursesState } from './types';

const initialState: CoursesState = {
  courses: [],
};

// ACTIONS
const getCourses = createAsyncThunk('courses/getCourses', async (_, { getState }) => {
  const res = await ax().post<{courses: Course[]}>('courses/api/course/list', {
    profile_id: getState().auth.profile_id,
  });
  const courses = res.data.courses;
  FastImage.preload(
    courses.filter(c => !!c.file_preview_url).map(c => ({ uri: c.file_preview_url })),
  );
  return courses;
});

// REDUCER
const slice = createSlice({
  name: 'courses',
  initialState,
  extraReducers: {
    [getCourses.fulfilled.type]: (state, action: PayloadAction<Course[]>) => {
      state.courses = action.payload;
    },
  },
  reducers: {},
});

// EXPORT
export * from './types';
export const coursesReducer = slice.reducer as Reducer<typeof initialState>;
export const coursesActions = { getCourses };
