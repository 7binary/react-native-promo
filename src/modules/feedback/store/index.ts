import { createAsyncThunk, createSlice, PayloadAction, Reducer } from '@reduxjs/toolkit';
import { ax } from '@pmk-team/common';
import { Faq, FeedbackMessage, FeedbackState } from './types';

const initialState: FeedbackState = {
  faqs: [],
  messages: [],
};

// ACTIONS
const getFaqs = createAsyncThunk('feedback/getFaqs', async () => {
  const res = await ax().post<{faqs: Faq[]}>('feedback/api/feedback/faq');
  return res.data.faqs;
});

const getFeedbackMessages = createAsyncThunk('feedback/getFeedbackMessages', async (_, { getState }) => {
  const res = await ax().post<{messages: FeedbackMessage[]}>('feedback/api/feedback/my-questions', {
    profile_id: getState().auth.profile_id,
  });
  const messages = res.data.messages;
  messages.sort((msg1: FeedbackMessage, msg2: FeedbackMessage): number => {
    const date1 = Date.parse(msg1.created_at);
    const date2 = Date.parse(msg2.created_at);
    if (date1 === date2) {
      return 0;
    }
    return date1 > date2 ? 1 : -1;
  });
  return messages;
});

// REDUCER
const slice = createSlice({
  name: 'feedback',
  initialState,
  extraReducers: {
    [getFaqs.fulfilled.type]: (state, action: PayloadAction<Faq[]>) => {
      state.faqs = action.payload;
    },
    [getFeedbackMessages.fulfilled.type]: (state, action: PayloadAction<FeedbackMessage[]>) => {
      state.messages = action.payload;
    },
  },
  reducers: {},
});

// EXPORT
export * from './types';
export const feedbackReducer = slice.reducer as Reducer<typeof initialState>;
export const feedbackActions = { getFaqs, getFeedbackMessages };
