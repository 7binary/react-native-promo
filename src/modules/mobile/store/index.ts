import {
  createAsyncThunk,
  createSelector,
  createSlice,
  PayloadAction,
  Reducer,
} from '@reduxjs/toolkit';
import firebase from 'react-native-firebase';
import { Dimensions, Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import { ax } from '@pmk-team/common';
import { MobileState, NotificationsResponse } from './types';

const initialState: MobileState = {
  notifications: [],
  unreaded: 0,
};

// ACTIONS
const getNotifications = createAsyncThunk('mobile/getNotifications', async (_, { getState }) => {
  const res = await ax().post<NotificationsResponse>('mobile/api/notifications', {
    profile_id: getState().auth.profile_id,
  });
  return res.data;
});

const readNotifications = createAsyncThunk('mobile/readNotifications', async (_, { getState }) => {
  const profile_id = getState().auth.profile_id;
  const notifications = getState().mobile.notifications;
  if (!profile_id || notifications.length === 0) {
    return;
  }
  const notification_ids = notifications.map(n => n.id);
  return ax().post('mobile/api/notifications/readed', { profile_id, notification_ids });
});

const registerFirebaseDevice = createAsyncThunk('mobile/registerFirebaseDevice', async (_, { getState }) => {
  await firebase.messaging().requestPermission();
  const token = await firebase.messaging().getToken();
  return ax().post('mobile/api/firebase/register-device', {
    profile_id: getState().auth.profile_id,
    token,
    platform: Platform.OS,
  });
});

const registerMobileDevice = createAsyncThunk('mobile/registerMobileDevice', async (_, { getState }) => {
  const manufacturer = await DeviceInfo.getManufacturer();
  const payload = {
    profile_id: getState().auth.profile_id,
    manufacturer,
    platform: Platform.OS,
    version: DeviceInfo.getVersion(),
    device_version: DeviceInfo.getSystemVersion(),
    device: DeviceInfo.getDeviceId(),
    screen_width: Math.round(Dimensions.get('window').width),
    screen_height: Math.round(Dimensions.get('window').height),
  };
  return ax().post('mobile/api/devices/add', payload);
});

// REDUCER
const slice = createSlice({
  name: 'mobile',
  initialState,
  extraReducers: {
    [getNotifications.fulfilled.type]: (state, action: PayloadAction<NotificationsResponse>) => {
      const { notifications, unreaded } = action.payload;
      state.notifications = notifications;
      state.unreaded = unreaded;
    },
  },
  reducers: {},
});

// EXPORT
export * from './types';
export const mobileReducer = slice.reducer as Reducer<typeof initialState>;
export const mobileActions = {
  getNotifications,
  readNotifications,
  registerFirebaseDevice,
  registerMobileDevice,
};

const selectSelf = (rootState: {mobile: MobileState}) => rootState.mobile;
export const unreadedSelector = createSelector(selectSelf, state => state.unreaded);
