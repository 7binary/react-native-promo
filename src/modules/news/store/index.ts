import { createAsyncThunk, createSlice, PayloadAction, Reducer } from '@reduxjs/toolkit';
import FastImage from 'react-native-fast-image';
import { ax } from '@pmk-team/common';
import { Instruction, NewsState, Publication } from './types';

const initialState: NewsState = {
  news: [],
  instructions: [],
};

// ACTIONS
const getNews = createAsyncThunk('news/getNews', async (_, { getState }) => {
  const res = await ax().post<{news: Publication[]}>('news/api/news', {
    profile_id: getState().auth.profile_id,
  });
  const { news } = res.data;
  FastImage.preload(
    news.filter(p => !!p.image_preview_url).map(p => ({ uri: p.image_preview_url })),
  );
  return news;
});

const getInstructions = createAsyncThunk('news/getInstructions', async (_, { getState }) => {
  const res = await ax().post<{instructions: Instruction[]}>('news/api/instructions', {
    profile_id: getState().auth.profile_id,
  });
  const { instructions } = res.data;
  FastImage.preload(
    instructions.filter(i => !!i.image_preview_url).map(i => ({ uri: i.image_preview_url })),
  );
  return instructions;
});

const readPublication = createAsyncThunk('news/readPublication',
  async (publication: Publication, { getState }) => {
    return ax().post('news/api/news/readed', {
      profile_id: getState().auth.profile_id,
      news_id: publication.id,
    });
  });

const readInstruction = createAsyncThunk('news/readInstruction',
  async (instruction: Instruction, { getState }) => {
    return ax().post('news/api/instructions/readed', {
      profile_id: getState().auth.profile_id,
      instruction_id: instruction.id,
    });
  });

// REDUCER
const slice = createSlice({
  name: 'news',
  initialState,
  extraReducers: {
    [getNews.fulfilled.type]: (state, action: PayloadAction<Publication[]>) => {
      state.news = action.payload;
    },
    [getInstructions.fulfilled.type]: (state, action: PayloadAction<Instruction[]>) => {
      state.instructions = action.payload;
    },
  },
  reducers: {},
});

// EXPORT
export * from './types';
export const newsReducer = slice.reducer as Reducer<typeof initialState>;
export const newsActions = {
  getNews,
  getInstructions,
  readPublication,
  readInstruction,
};
