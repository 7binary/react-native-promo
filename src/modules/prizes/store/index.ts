import {
  createAsyncThunk,
  createSelector,
  createSlice,
  PayloadAction,
  Reducer,
} from '@reduxjs/toolkit';
import FastImage from 'react-native-fast-image';
import { ax } from '@pmk-team/common';
import {
  Cart,
  CartPositon,
  CatalogCard,
  CatalogOrder,
  Payment,
  PaymentSettings,
  PrizesState,
  ShopOrder,
  ShopProduct,
} from './types';

const initialState: PrizesState = {
  paymentSettings: [],
  payments: [],
  catalogCards: [],
  catalogOrders: [],
  shopProducts: [],
  shopOrders: [],
  cart: {
    positions: [],
    length: 0,
    summary: 0,
  },
};

// ACTIONS
const getPaymentSettings = createAsyncThunk('prizes/getPaymentSettings', async () => {
  const res = await ax().post<{settings: PaymentSettings[]}>('payments/api-v3/settings/view');
  return res.data.settings;
});

const getPayments = createAsyncThunk('prizes/getPayments', async (_, { getState }) => {
  const res = await ax().post<{payments: Payment[]}>('payments/api-v3/payments/by-profile', {
    profile_id: getState().auth.profile_id,
  });
  return res.data.payments;
});

const getCatalogCards = createAsyncThunk('prizes/getCatalogCards', async () => {
  const res = await ax().post<{cards: CatalogCard[]}>('catalog/api-v3/cards/list');
  const cards = res.data.cards;
  FastImage.preload(
    cards.filter(c => !!c.image).map(c => ({ uri: c.image })),
  );
  return cards;
});

const getCatalogOrders = createAsyncThunk('prizes/getCatalogOrders', async (_, { getState }) => {
  const res = await ax().post<{orders: CatalogOrder[]}>('catalog/api-v3/users/orders', {
    profile_id: getState().auth.profile_id,
  });
  return res.data.orders;
});

const getShopProducts = createAsyncThunk('prizes/getShopProducts', async (_, { getState }) => {
  const res = await ax().post<{products: ShopProduct[]}>('shop/api-v3/product/list', {
    profile_id: getState().auth.profile_id,
  });
  return res.data.products;
});

const getShopOrders = createAsyncThunk('prizes/getShopOrders', async (_, { getState }) => {
  const res = await ax().post<{orders: ShopOrder[]}>('shop/api-v3/order/list', {
    profile_id: getState().auth.profile_id,
  });
  return res.data.orders;
});

// REDUCER
const slice = createSlice({
  name: 'prizes',
  initialState,
  extraReducers: {
    [getPaymentSettings.fulfilled.type]: (state, action: PayloadAction<PaymentSettings[]>) => {
      state.paymentSettings = action.payload;
    },
    [getPayments.fulfilled.type]: (state, action: PayloadAction<Payment[]>) => {
      state.payments = action.payload;
    },
    [getCatalogCards.fulfilled.type]: (state, action: PayloadAction<CatalogCard[]>) => {
      state.catalogCards = action.payload;
    },
    [getCatalogOrders.fulfilled.type]: (state, action: PayloadAction<CatalogOrder[]>) => {
      state.catalogOrders = action.payload;
    },
    [getShopProducts.fulfilled.type]: (state, action: PayloadAction<ShopProduct[]>) => {
      state.shopProducts = action.payload;
    },
    [getShopOrders.fulfilled.type]: (state, action: PayloadAction<ShopOrder[]>) => {
      state.shopOrders = action.payload;
    },
  },
  reducers: {
    resetCart: (state, action: PayloadAction<'card' | 'product'>) => {
      if (action.payload === 'card') {
        state.cart.positions = state.cart.positions.filter(pos => pos.card == null);
      } else if (action.payload === 'product') {
        state.cart.positions = state.cart.positions.filter(pos => pos.product == null);
      }
      state.cart = calculateCart(state.cart);
    },
    changeCartQty: (state, action: PayloadAction<{position: CartPositon, qty: number}>) => {
      const { qty, position } = action.payload;
      const { card, product, nominal } = position;
      let cart = state.cart;

      for (let i = 0; i < cart.positions.length; i++) {
        const positionCard = cart.positions[i].card;
        const positionProduct = cart.positions[i].product;
        //  если добавляем/удаляем продукт
        if (positionProduct && product && positionProduct.id === product.id) {
          cart.positions[i].qty += qty;
          if (cart.positions[i].qty <= 0) {
            cart.positions = cart.positions.filter((pos: CartPositon) => {
              if (!pos.product) {return true;}
              if (pos.product.id !== cart.positions[i]?.product?.id) {return true;}
              return false;
            });
          }
        }
        //  если добавляем/удаляем сертификат
        if (positionCard && card && positionCard.type === card.type && cart.positions[i].nominal === nominal) {
          cart.positions[i].qty += qty;
          if (cart.positions[i].qty <= 0) {
            cart.positions = cart.positions.filter((pos: CartPositon) => {
              if (!pos.card
                || pos.card.type !== cart.positions[i]?.card?.type
                || pos.nominal !== cart.positions[i].nominal) {
                return true;
              }
              return false;
            });
          }
          break;
        }
      }
      state.cart = calculateCart(cart);
    },
    addToCart: (state, action: PayloadAction<CartPositon>) => {
      const position = action.payload;
      const { card, product, nominal, qty } = position;
      let cart = state.cart;
      let hasPosition = false;

      for (let i = 0; i < cart.positions.length; i++) {
        const positionCard = cart.positions[i].card;
        const positionProduct = cart.positions[i].product;
        // проверяем что такой товар уже лежит в корзине
        if (positionProduct && product && positionProduct.id === product.id) {
          cart.positions[i].qty += qty;
          hasPosition = true;
          break;
        }
        // проверяем что такой сертификат уже лежит в корзине
        if (positionCard && card && positionCard.type === card.type && cart.positions[i].nominal === nominal) {
          cart.positions[i].qty += qty;
          hasPosition = true;
          break;
        }
      }
      if (!hasPosition) {
        cart.positions.push(position);
      }
      state.cart = calculateCart(cart);
    },
  },
});
const { resetCart, changeCartQty, addToCart } = slice.actions;

// HELPERS
function calculateCart(cartOrig: Cart): Cart {
  const cart = { ...cartOrig };
  let cartLength = 0;
  let cartSummary = 0;

  for (let i = 0; i < cart.positions.length; i++) {
    cartLength += cart.positions[i].qty;
    if (cart.positions[i].cardNominal) {
      cartSummary += cart.positions[i].qty * cart.positions[i].cardNominal!.price;
    } else {
      cartSummary += cart.positions[i].qty * cart.positions[i].nominal;
    }
  }

  cart.length = cartLength;
  cart.summary = cartSummary;

  return cart;
}

// EXPORT
export * from './types';
export const prizesReducer = slice.reducer as Reducer<typeof initialState>;
export const prizesActions = {
  resetCart,
  changeCartQty,
  addToCart,
  getPaymentSettings,
  getPayments,
  getCatalogCards,
  getCatalogOrders,
  getShopProducts,
  getShopOrders,
};

const selectSelf = (rootState: {prizes: PrizesState}) => rootState.prizes;
export const cartSelector = createSelector(selectSelf, state => state.cart);

