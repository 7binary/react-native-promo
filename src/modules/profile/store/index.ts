import {
  createAsyncThunk,
  createSelector,
  createSlice,
  PayloadAction,
  Reducer,
} from '@reduxjs/toolkit';
import FastImage from 'react-native-fast-image';
import { ax, Page } from '@pmk-team/common';
import { Banner, Passport, ProfileState, Transaction, User } from './types';

const initialState: ProfileState = {
  user: null,
  transactions: [],
  passport: null,
  banners: [],
  rulesPage: null,
};

// ACTIONS
const getProfile = createAsyncThunk('profile/getProfile', async (_, { getState }) => {
  const res = await ax().post<{profile: User}>('profiles/api/auth/get-profile', {
    profile_id: getState().auth.profile_id,
  });
  return res.data.profile;
});

const getTransactions = createAsyncThunk('profile/getTransactions', async (_, { getState }) => {
  const res = await ax().post<{transactions: Transaction[]}>('profiles/api/transaction/list', {
    profile_id: getState().auth.profile_id,
  });
  return res.data.transactions;
});

const getPassport = createAsyncThunk('profile/getPassport', async (_, { getState }) => {
  const res = await ax().post<{form: Passport | null}>('taxes/api/passport', {
    profile_id: getState().auth.profile_id,
  });
  return res.data.form;
});

const getBanners = createAsyncThunk('profile/getBanners',
  async (group: string = 'guest-index') => {
    const res = await ax().post<{banners: Banner[]}>('posters/api/posters/by-group', { group });
    FastImage.preload(
      res.data.banners.filter(b => !!b.mobile_banner_url).map(b => ({ uri: b.mobile_banner_url })),
    );
    return res.data.banners;
  });

const getRules = createAsyncThunk('profile/getRules', async () => {
  const res = await ax().post<{page: Page | null}>('pages/api/pages', { url: 'rules' });
  return res.data.page;
});

// REDUCER
export const slice = createSlice({
  name: 'profile',
  initialState,
  extraReducers: {
    [getProfile.fulfilled.type]: (state, action: PayloadAction<User>) => {
      state.user = action.payload;
    },
    [getTransactions.fulfilled.type]: (state, action: PayloadAction<Transaction[]>) => {
      state.transactions = action.payload;
    },
    [getPassport.fulfilled.type]: (state, action: PayloadAction<Passport | null>) => {
      state.passport = action.payload;
    },
    [getBanners.fulfilled.type]: (state, action: PayloadAction<Banner[]>) => {
      state.banners = action.payload;
    },
    [getRules.fulfilled.type]: (state, action: PayloadAction<Page | null>) => {
      state.rulesPage = action.payload;
    },
  },
  reducers: {
    setUser: (state, action: PayloadAction<User>) => {
      state.user = action.payload;
    },
    unsetUser: state => {
      state.user = null;
      state.passport = null;
      state.transactions = [];
    },
  },
});
const { setUser, unsetUser } = slice.actions;

// EXPORT
export * from './types';
export const profileReducer = slice.reducer as Reducer<typeof initialState>;
export const profileActions = {
  setUser,
  unsetUser,
  getProfile,
  getPassport,
  getBanners,
  getTransactions,
  getRules,
};

const selectSelf = (rootState: {profile: ProfileState}) => rootState.profile;
export const userSelector = createSelector(selectSelf, state => state.user);
export const passportSelector = createSelector(selectSelf, state => state.passport);
