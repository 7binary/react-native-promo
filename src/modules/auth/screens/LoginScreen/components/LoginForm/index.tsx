import React from 'react';
import { View } from 'react-native';
import { Field, Formik } from 'formik';
import { Button } from 'react-native-paper';

import { FormInput, ErrorMessage, Settings, ax } from '@pmk-team/common';
import { useActions } from 'store';

const LoginForm = () => {
  const { login } = useActions();

  return (
    <Formik
      initialValues={{ login: '', password: '' }}
      onSubmit={(values, actions) => {
        login({
          client: ax({ actions }),
          values,
          onSuccess: () => actions.resetForm(),
        });
      }}
    >
      {({ handleSubmit, isValid, isSubmitting, status }) => (
        <>
          <Field
            component={FormInput}
            name="login"
            label="Номер телефона"
            iconName="ios-call-outline"
            keyboardType="numeric"
            returnKeyType="done"
            phone
          />
          <Field
            component={FormInput}
            name="password"
            label="Пароль"
            iconName="ios-lock-closed-outline"
            secureTextEntry
          />
          <ErrorMessage dot>{status}</ErrorMessage>
          <View>
            <Button
              mode="contained"
              onPress={handleSubmit}
              disabled={!isValid || isSubmitting}
              loading={isSubmitting}
              color={Settings.colors.primary}
            >Войти</Button>
          </View>
        </>
      )}
    </Formik>
  );
};

export default LoginForm;
