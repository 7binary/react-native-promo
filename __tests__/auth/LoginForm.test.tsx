import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import LoginForm from 'modules/auth/screens/LoginScreen/components/LoginForm';
import { store } from 'store';

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <LoginForm/>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
