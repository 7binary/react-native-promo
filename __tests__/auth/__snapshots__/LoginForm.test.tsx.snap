// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "marginBottom": 5,
      }
    }
  >
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#444",
              "fontSize": 28,
            },
            Object {
              "left": 0,
              "padding": 12,
              "paddingLeft": 20,
              "position": "absolute",
              "top": 0,
              "zIndex": 1000,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(231, 231, 231)",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
            Object {
              "paddingLeft": 45,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.26)",
              "bottom": 0,
              "height": 2,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "scaleY": 0.5,
                },
              ],
            }
          }
        />
        <View
          style={
            Object {
              "minHeight": 52,
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 3,
                  },
                ],
              }
            }
          >
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 28,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -10,
                    },
                    Object {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Номер телефона
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 28,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -10,
                    },
                    Object {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Номер телефона
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            keyboardType="numeric"
            multiline={false}
            numberOfLines={4}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            options={
              Object {
                "mask": "+7 (999) 999-9999",
              }
            }
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            returnKeyType="done"
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "zIndex": 1,
                },
                Object {
                  "paddingLeft": 12,
                  "paddingRight": 12,
                },
                Object {
                  "height": 52,
                },
                Object {
                  "paddingBottom": 2,
                  "paddingTop": 22,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            type="custom"
            underlineColorAndroid="transparent"
            value="+7"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#bf0f17",
            "fontSize": 12,
            "marginBottom": 0,
            "marginLeft": 10,
            "minHeight": 15,
          }
        }
      />
    </View>
  </View>,
  <View
    style={
      Object {
        "marginBottom": 5,
      }
    }
  >
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#444",
              "fontSize": 28,
            },
            Object {
              "left": 0,
              "padding": 12,
              "paddingLeft": 20,
              "position": "absolute",
              "top": 0,
              "zIndex": 1000,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(231, 231, 231)",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
            Object {
              "paddingLeft": 45,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.26)",
              "bottom": 0,
              "height": 2,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "scaleY": 0.5,
                },
              ],
            }
          }
        />
        <View
          style={
            Object {
              "minHeight": 52,
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 28,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Пароль
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 28,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Пароль
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            multiline={false}
            numberOfLines={4}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            secureTextEntry={true}
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "zIndex": 1,
                },
                Object {
                  "paddingLeft": 12,
                  "paddingRight": 12,
                },
                Object {
                  "height": 52,
                },
                Object {
                  "paddingBottom": 2,
                  "paddingTop": 22,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#bf0f17",
            "fontSize": 12,
            "marginBottom": 0,
            "marginLeft": 10,
            "minHeight": 15,
          }
        }
      />
    </View>
  </View>,
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#333",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            Войти
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;
